test/x3/char_set.cpp

```diff
@@ -0,0 +1,89 @@
+#define BOOST_SPIRIT_X3_UNICODE
+
+#include <boost/detail/lightweight_test.hpp>
+#include <boost/spirit/home/x3.hpp>
+#include <boost/spirit/home/support/char_encoding/ascii.hpp>
+#include <boost/spirit/home/support/char_encoding/unicode.hpp>
+#include <boost/type_traits/is_same.hpp>
+
+#include "test.hpp"
+#include <vector>
+#include <algorithm>
+#include <boost/utility/string_view.hpp>
+
+int
+main()
+{
+    using spirit_test::test;
+
+    // ascii
+    {
+        using namespace boost::spirit::x3;
+        using char_set = char_set<boost::spirit::char_encoding::ascii>;
```

Comment:
Ain't we already have those tests?

Reply:
> Ain't we already have those tests?

Good question! I was thinking about that actually. Was it in char1.cpp? Perhaps we can just augment those?

Reply:
Ah, Kojoley is correct.

>  … if you did not know it 
At the time, I didn't. You are correct, the ASCII tests are repetitious in this case.

Should I simply move the Unicode tests here to the char1.cpp test file?

Reply:
> At the time, I didn't. You are correct, the ASCII tests are repetitious in this case.
> 
> Should I simply move the Unicode tests here to the char1.cpp test file?

Yes please :-)

Reply:
Done!